src/Strategy/Node/Pnpm/PnpmLock.hs

```diff
@@ -76,17 +77,55 @@ import Path (Abs, File, Path)
 --          - For dependency resolved via git resolver, format is: "${Url}".
 --          - For dependency resolved via directory resolver, format is: "file:${relativePath}".
 --
+--
+--  Pnpm lockfile (v6) differs (v5), in following manner:
+--  -----------------------------------------------------
+--
+--    * `importers` shape merges specifiers and version, in singular object:
+--      @
+--      > importers:
+--      >    dependencies:
+--      >      aws-sdk:
+--      >        specifier: 2.1148.0
+--      >        version: 2.1148.0
+--      @
+--
+--    * Key of `packages` refer (e.g. "/buffer@4.9.2") denotes name of dependency and resolved version using '@' separator
+--        - For dependency resolved via registry resolver, format is: "/${dependencyName}@${resolvedVersion}${peerDepsInBrackets}".
+--      @
+--      >   /ieee754@1.1.13:
+--      >      resolution: {integrity: sha512...}
+--      >      dev: false
+--      >
+--      >   /@clerk/nextjs@4.22.1(next@13.4.10)(react-dom@18.2.0)(react@18.2.0):
+--      >       resolution: {integrity: sha512...}
+--      @
+--
+--    * If project has set peerDependencies to be not auto installed, pnpm
+--      by default, does not include them in the lockfile. So, no additional
+--      work is required for newly introduced `settings.autoInstallPeers` field.
+--
 --  References:
 --    - [pnpm](https://pnpm.io/)
 --    - [pnpm-lockfile](https://github.com/pnpm/pnpm/blob/5cfd6d01946edcce86f62580bddc788d02f93ed6/packages/lockfile-types/src/index.ts)
+--    - [pnpm-lockfile-v6](https://github.com/pnpm/pnpm/pull/5810/files)
 data PnpmLockfile = PnpmLockfile
   { importers :: Map Text ProjectMap
   , packages :: Map Text PackageData
+  , lockFileVersion :: PnpmLockFileVersion
   }
   deriving (Show, Eq, Ord)
 
+data PnpmLockFileVersion = PnpmLock5 | PnpmLock6 deriving (Show, Eq, Ord)
+
 instance FromJSON PnpmLockfile where
   parseJSON = Yaml.withObject "pnpm-lock content" $ \obj -> do
+    rawLockFileVersion :: TextLike <- obj .:? "lockfileVersion" .!= (TextLike mempty)
+    let lockFileVersion =
+          if Text.isPrefixOf "6" (unTextLike rawLockFileVersion)
+            then PnpmLock6
+            else PnpmLock5
```

Comment:
This construction means that we'll parse lockfiles at v4 and below like v5, but it also means we'll parse lockfiles at v7 and above like v5 which seems incorrect.

Please compare the version in a way that properly accounts for this, unless there is a good reason for this functionality.